### PR TITLE
Validate URL pattern that is passed to watch method

### LINKED ISSUE #122: If invalid URL pattern is provided to watch() method, reject promise with SyntaxError
At the moment, if invalid URL pattern is passed to watch() method, API provides no indication about this error condition. It would be good to inform developer immediately, otherwise 'dispatching NFC content' would not work without valid reason.

### PR DESCRIPTION
**Pull request by alexshalamov**
Fixes #122

### CODE REVIEW

**Comment by alexshalamov** [comment]
@zolkis @kenchris could you please take a look?

**Comment by zolkis** [comment]
When we say "validate", I think we need to link to an algorithm.

File: index.html [hunk 1]
```diff
@@ -2495,6 +2495,12 @@
             <code>"NotSupportedError"</code> and abort these steps.
           </li>
           <li>
+            If the <var>options</var>.url is not an empty string, validate
```
Comment (kenchris) [inline index.html:2498]: Shouldnt you link to how to validate?

Comment (alexshalamov) [reply index.html:2498]: Sure, would add validation algorithm.

**Comment by alexshalamov** [comment]
@zolkis @kenchris how strict (complex) validation should be? Since URL pattern is not a valid URL, do we want to re-define URL parser in the spec?

**Comment by kenchris** [comment]
It probably has to be quite strict. Maybe we could monkey patch? @anssiko

**Comment by alexshalamov** [comment]
@kenchris then we need to write ABNF for 'NFC URL pattern' that should be as complex as the one for URL's

At the moment, valid 'NFC URL pattern' could be https://site.com:test/path , while this is not a valid URL.

**Comment by kenchris** [comment]
Yeah, for interoperability this is probably what we need to do. Not sure if there are any shortcuts or it something similar is done elsewhere in the platform

**Comment by alexshalamov** [comment]
Created #124, same functionality could be achieved without rewriting URL parser or defining custom ABNF notation.

**Comment by anssiko** [comment]
You might be able to use https://url.spec.whatwg.org/#url-parsing

**Comment by alexshalamov** [comment]
@anssiko yes, if we simplify nfc url patterns (#124). With current ABNF notation, url parsers would fail to validate / parse the pattern.

**Comment by zolkis** [comment]
Let's go with simplifying patterns. This is a good example why discipline/harmonization of web specs is needed, now we could reuse an existing algorithm.

**Pull request closed without merging**